src/auth/src/credentials/external_account.rs

```diff
@@ -25,13 +26,31 @@ use crate::{BuildResult, Result};
 use http::{Extensions, HeaderMap};
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
+use std::future::Future;
 use std::sync::Arc;
 use tokio::time::{Duration, Instant};
 
-#[async_trait::async_trait]
-pub(crate) trait SubjectTokenProvider: std::fmt::Debug + Send + Sync {
-    /// Generate subject token that will be used on STS exchange.
-    async fn subject_token(&self) -> Result<String>;
+pub trait SubjectTokenProvider: std::fmt::Debug + Send + Sync {
+    fn subject_token(&self) -> impl Future<Output = Result<String>> + Send;
+}
```

Comment:
So to override this I need to implement my own client, error detection, parsing, etc.?  Bummer.

Reply:
I think the main point of this custom subject token provider is indeed for the customer own all the stack around fetching the subject token, specially in complex scenarios like the ones we discussed where the customers needs to use a custom openssl version and other details. I think we are also evaluating how to allow injecting custom headers and/or provide a custom http client cc @sai-sunder-s